### DISCUSSION TITLE
blog/how-to-build-shopware-extension-assets-faster/

### DISCUSSION THREAD

**Discussion by giscus** [shyim/shyim.me#47] [Q&A]
# blog/how-to-build-shopware-extension-assets-faster/

When you deliver your Shopware extension to a Shop owner, your extension must contain in src/Resources/public the build assets for the Administration and the Storefront. We committed in the first plugins in Frosh all assets directly in Git. But this led to annoying conflicts, or when you merged two pull requests at once, it didn’t contain both changes.
So, for this reason, we removed the build files from all Git repositories again and used shopware-cli to build the files.

https://shyim.me/blog/how-to-build-shopware-extension-assets-faster/

**Comment by JuicyLung91** [comment]
Nice. I will check this out. Does the build also work with a dockware:dev setup where you mount your plugin custom/plugins/PluginName directory?

**Reply by shyim** [reply]
It should work on any system except systems based alpine linux

**Reply by JuicyLung91** [reply]
Hmm for me the vue dev tools does not work with the admin watcher. It says it is not in dev mode. Any way to set the mode? The menv file has ist already in dev.

**Reply by JuicyLung91** [reply]
I mean the env file has development mod. Hmm can you confirm that vue dev tools does not work with this watcher?

**Comment by ioanok** [comment]
Does the build takes more time for the first time? Seems that it ignores the ESBuild configs from `.shopware-extension.yml` and installs node stuff. 
I tried it for the entire project: 
`shopware-cli project admin-build` with `.shopware-extension.yml` in project root 
- 1st run takes about **2min 50s**,
- 2nd run takes about **9s** to build

Any idea why? Or am I doing something wrong?

**Reply by shyim** [reply]
Can you show me your `shopware-extension.yml` ?

**Reply by ioanok** [reply]
It's the same as on the article
````yaml
build:
  zip:
    assets:
      enable_es_build_for_admin: true
      enable_es_build_for_storefront: true

````

**Reply by ioanok** [reply]
I just investiagated the repo a bit and seems that ESBuild is not supported by `project` section of the tool. Can you confirm that?
